myia/abstract/data.py

```diff
@@ -860,6 +860,17 @@ def listof(t):
 def u64tup_typecheck(engine, tup):
     """Verify that tup is a tuple of uint64."""
     tup_t = engine.check(AbstractTuple, tup)
+    # If tup is a constant tuple of positive integers,
+    # then consider it as a tuple of uint64.
```

Comment:
That's probably not going to cause any issues, but I'm a bit hesitant that this warrants a special case.

Reply:
Ok, I will remove it.